src/directives/async-append.ts

```diff
@@ -67,7 +67,9 @@ export const asyncAppend = directive(
         // we accept a mapper function. This is especially convenient for
         // rendering a template for each item.
         if (mapper !== undefined) {
-          v = mapper(v, i);
+          // This is safe because T must otherwise be treated as unknown by
+          // the rest of the system.
+          v = mapper(v, i) as T;
```

Comment:
I really wish TypeScript had a way to re-declare the type of a variable for these purposes.

Reply:
Same.